### ISSUE TITLE
Opencv 3.4.1   Cannot load Tensorflow Model(Unknown layer type Shape in op dropout/Shape)

### ISSUE THREAD

**Issue by Mainvooid** [opencv/opencv#11512]
##### System information (version)
- OpenCV => 3.4.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2017

### Detailed description
I was using tf.nn.dropout.
#### Steps to reproduce
**Save model**
```python
constant_graph = graph_util.convert_variables_to_constants(sess, sess.graph_def, ["accuracy"])
with tf.gfile.FastGFile("./future_model/p_model_save.pb", mode='wb') as f:
    f.write(constant_graph.SerializeToString())
```
**Load the saved model in tensorflow and run it successfully**
```python
with tf.Session() as sess:
    with tf.gfile.FastGFile("./future_model/p_model_save.pb", 'rb') as f:
        graph_def = tf.GraphDef()
        graph_def.ParseFromString(f.read()) 

        input_x,input_y,keep_prob,accuracy=tf.import_graph_def(graph_def, return_elements=["input_x:0","input_y:0","keep_prob:0","accuracy:0"])
        print(input_x,input_y,keep_prob,accuracy)
   
    sess.run(tf.global_variables_initializer())

    print(sess.run(accuracy,  feed_dict={input_x: pre_input_x, input_y: pre_input_y,keep_prob:1.0}))
```
**Load the saved model in opencv, but run failure.**
```cpp
int main(int argc, char **argv)
{
	String modelFile = "./future_model/p_model_save.pb";
	cout << "readNetFromTensorflow" << endl;
	dnn::Net net = readNetFromTensorflow(modelFile);
	cout << "read success" << endl;
}
```
**Error message**
```
readNetFromTensorflow

dropout/Shape:Shape(relu_layer)
out_type:[ ]
T:0
OpenCV(3.4.1) Error: Unspecified error (Unknown layer type Shape in op dropout/Shape) in cv::dnn::experimental_dnn_v4::`anonymous-namespace'::TFImporter::populateNet, file C:\build\master_winpack-build-win64-vc15\opencv\modules\dnn\src\tensorflow\tf_importer.cpp, line 1582
```

**Comment by dkurt** [comment]
@Mainvooid, please try with the latest state of OpenCV: https://github.com/opencv/opencv/tree/3.4.

**Comment by dkurt** [comment]
@Mainvooid, `tf.nn.dropout` has no testing mode which means it always generates random values and for similar inputs you'll get different outputs. Do you really want to create a not deterministic model?

**Comment by dkurt** [comment]
@Mainvooid, you need to save graph with `keep_prob:1.0` or without dropout because there is already serialized `keep_prob` at graph and I guess it has different value.

**Comment by dkurt** [comment]
Close an issue due no response from author.
